pageserver/src/page_service.rs

```diff
@@ -376,32 +380,68 @@ impl PageServerHandler {
             // TODO: We could create a new per-request context here, with unique ID.
             // Currently we use the same per-timeline context for all requests
 
+            // Remotexact: We assume that a relation can only be moved once from the main region,
+            // where it is created, to a new region.
+            // After the move, timeline of the new region does not know about the existence of the
+            // pages of the relation, resulting in an error in the pageserver (even if the page is empty).
+            // We thus fallback to the main region in that case.
+            // Note that the compute node generally only logs the changes on the page, not the whole page.
+            // This is why the relation must be empty when it is moved, otherwise the new region will lose
+            // the data added to the relation prior to the move.
             let response = match neon_fe_msg {
-                PagestreamFeMessage::Exists(req) => {
+                PagestreamFeMessage::Exists(mut req) => {
                     let _timer = metrics.get_rel_exists.start_timer();
                     match get_timeline_by_region_id(&timelines, req.region) {
                         Ok(timeline) => {
-                            self.handle_get_rel_exists_request(timeline.as_ref(), &req, &ctx)
+                            match self
+                                .handle_get_rel_exists_request(timeline.as_ref(), &req, &ctx)
                                 .await
+                            {
+                                res @ Ok(_) => res,
+                                Err(_) => {
+                                    req.latest = true;
+                                    req.lsn = Lsn(0);
```

Comment:
Why do we update the LSN's to 0 here and elsewhere? Shouldn't it be the latest LSN of the main region?